tests/cephfs/cephfs_mirroring/test_cephfs_mirror_auth_caps.py

```diff
@@ -0,0 +1,523 @@
+        target_site_name = "remote_site_caps"
+
+        log.info("Enable mirroring mgr module on source and destination")
+        enable_mirroring_on_source = fs_mirroring_utils.enable_mirroring_module(
+            source_clients[0]
+        )
+        if enable_mirroring_on_source:
+            log.error("Mirroring module not enabled on Source Cluster.")
+            raise CommandFailed("Enable mirroring mgr module failed")
+
+        enable_mirroring_on_target = fs_mirroring_utils.enable_mirroring_module(
+            target_clients[0]
+        )
+        if enable_mirroring_on_target:
+            log.error("Mirroring module not enabled on Target Cluster.")
+            raise CommandFailed("Enable mirroring mgr module failed")
+        log.info(
+            "Scenario 1: Create users with r, rw ,rwp on mds and see token creation fails"
+        )
+        perform_fs_mirroring_operations(
+            fs_mirroring_utils,
+            target_fs,
+            target_user,
+            target_clients,
+            source_fs,
+            source_clients,
+            target_site_name,
+            permissions="r",
+        )
+        perform_fs_mirroring_operations(
+            fs_mirroring_utils,
+            target_fs,
+            target_user,
+            target_clients,
+            source_fs,
+            source_clients,
+            target_site_name,
+            permissions="rw",
+        )
+        perform_fs_mirroring_operations(
+            fs_mirroring_utils,
+            target_fs,
+            target_user,
+            target_clients,
+            source_fs,
+            source_clients,
+            target_site_name,
+            permissions="rwp",
+        )
+        log.info("Scenario 2: Update user auths and try to create bootstrap token")
+        log.info("Create a user on target cluster for peer connection")
+        target_user_for_peer = fs_mirroring_utils.create_authorize_user(
+            target_fs, target_user, target_clients[0]
+        )
+        if target_user_for_peer:
+            log.error("User Creation Failed with the expected caps")
+            raise CommandFailed("User Creation Failed")
+
+        log.info(f"Enable cephfs mirroring module on the {source_fs}")
+        fs_mirroring_utils.enable_snapshot_mirroring(source_fs, source_clients[0])
+
+        log.info("Create the peer bootstrap")
+        bootstrap_token = fs_mirroring_utils.create_peer_bootstrap(
+            target_fs, target_user, target_site_name, target_clients[0]
+        )
+        out, rc = target_clients[0].exec_command(
+            sudo=True, cmd=f"ceph auth get client.{target_user}"
+        )
+        log.info(out)
+        log.info("Before Modifying the caps")
+        caps_list = ["r", "rw", "rwp"]
+        for caps in caps_list:
+            target_clients[0].exec_command(
+                sudo=True,
+                cmd=f'ceph auth caps client.{target_user} mds "allow {caps} fsname={target_fs}" '
```

Comment:
test mentions about limited access top mds and osd, "Modify the user capabilities with limited access to the mds and osd", but we are modifying caps of just mds, not osd?